### PR TITLE
Documenting preemption toleration annotation for PriorityClasses

### PR DESCRIPTION
**Pull request by nitishchauhan0022**
Fixes: #37589

### CODE REVIEW

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 1]
```diff
@@ -547,6 +547,19 @@ This annotation requires the [NodePreferAvoidPods scheduling plugin](/docs/refer
 to be enabled. The plugin is deprecated since Kubernetes 1.22.
 Use [Taints and Tolerations](/docs/concepts/scheduling-eviction/taint-and-toleration/) instead.
 
+### preemption-toleration.scheduling.sigs.k8s.io/minimum-preemptable-priority
+
+Example: `preemption-toleration.scheduling.sigs.k8s.io/minimum-preemptable-priority: "10000"`
+
+This specifies the minimum priority value that can preempt this [PriorityClass](/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass). It defaults to the PriorityClass's priority value + 1 if not set, which means pods that have a higher priority value can preempt it.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:554]: - Is this an annotation or a label?
- What kind of objects is this used on?

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 2]
```diff
@@ -547,6 +547,19 @@ This annotation requires the [NodePreferAvoidPods scheduling plugin](/docs/refer
...
+
+### preemption-toleration.scheduling.sigs.k8s.io/toleration-seconds
+
+Example: `preemption-toleration.scheduling.sigs.k8s.io/toleration-seconds: "3600"`
+
+This specifies how long the [PriorityClass](/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass) can tolerate preemption
+by priorities lower than `MinimumPreemptablePriority`. It defaults to zero if not set. Zero value means the pod will be preempted immediately. i.e., no toleration at all. If it's set to a positive value, the duration will be honored. If it's set to a negative value, the pod can be tolerated forever - i.e., pods with priority lower than `MinimumPreemptablePriority` won't be able to preempt it. This value affects scheduled pods only (no effect on nominated pods).
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:561]: - Is this an annotation or a label?
- What kind of objects is this used on?
- (nit) [Avoid using Latin phrases](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-latin-phrases) and their abbreviations

**Review by sftim — COMMENTED** [review]
Thanks @nitishchauhan0022 

Would you be willing to make some more tweaks here? I've added inline comments about what's not clear and what's different from our style guide.

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/reference/labels-annotations-taints/_index.md:561]: Readers may also wonder what `MinimumPreemptablePriority` is. Can we help people to work that out?

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 1 above)
Comment (nitishchauhan0022) [reply content/en/docs/reference/labels-annotations-taints/_index.md:554]: - This is an annotaion, you also mentioned it while opening the issue. Should i add something like "This annotation specifies ___" to make it more clear.
- This is used on PriorityClass object, then the PriorityClass object is used for defining pod configuration. So, 
`Used On: PriorityClass` sound accurate to me.

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 2 above)
Comment (nitishchauhan0022) [reply content/en/docs/reference/labels-annotations-taints/_index.md:561]: `MinimumPreemptablePriority` is the annotation that is defined just above for PriorityClass, we can link this annotaion here but as it is defined just above so linking it doesn't making much sense to me. What do you think?

Comment (nitishchauhan0022) [reply content/en/docs/reference/labels-annotations-taints/_index.md:561]: > * (nit) [Avoid using Latin phrases](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-latin-phrases) and their abbreviations
ok, i will follow it.

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/reference/labels-annotations-taints/_index.md:554]: > Should i add something like "This annotation specifies ___" to make it more clear.

Yes

You should add Used On text too; see other entries for an example.

**Comment by nitishchauhan0022** [comment]
@sftim Please, take a look.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 3]
```diff
@@ -547,6 +547,23 @@ This annotation requires the [NodePreferAvoidPods scheduling plugin](/docs/refer
...
+This annotation specifies the minimum priority value that can preempt the [PriorityClass](/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass). It defaults to the PriorityClass's priority value +1, means pods that have a higher priority value can preempt it.
+
+### preemption-toleration.scheduling.sigs.k8s.io/toleration-seconds
+
+Example: `preemption-toleration.scheduling.sigs.k8s.io/toleration-seconds: "3600"`
+
+Used on: PriorityClass
+
+This annotation specifies how long the [PriorityClass](/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass) can tolerate preemption by priorities lower than `MinimumPreemptablePriority`. It defaults to value 0, means the pod will be preempted immediately that is, no toleration. If
+set to a positive value, the duration will be honored. If set to a negative value, the pod can be tolerated forever that is, pods with a priority lower than `MinimumPreemptablePriority` won't be able to preempt. This affects scheduled pods only (no effect on nominated pods).
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:565]: You should change `MinimumPreemptablePriority` to the actual key of that annotation: `preemption-toleration.scheduling.sigs.k8s.io/minimum-preemptable-priority`. For the second use, the English phrase “minimum preemptable priority“ is fine too.

**Comment by Kartik494** [comment]
@nitishchauhan0022  could you update issue description to `Fixes #issuenumber` , it will close the issue when this PR merged

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 3 above)
Comment (nitishchauhan0022) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: @sftim Done

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 4]
```diff
@@ -547,6 +547,22 @@ This annotation requires the [NodePreferAvoidPods scheduling plugin](/docs/refer
 to be enabled. The plugin is deprecated since Kubernetes 1.22.
 Use [Taints and Tolerations](/docs/concepts/scheduling-eviction/taint-and-toleration/) instead.
 
+### preemption-toleration.scheduling.sigs.k8s.io/minimum-preemptable-priority
+
+Example: `preemption-toleration.scheduling.sigs.k8s.io/minimum-preemptable-priority: "10000"`
+
+Used on: PriorityClass
+
+This annotation specifies the minimum priority value that can preempt the [PriorityClass](/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass). It defaults to the PriorityClass's priority value +1, means pods that have a higher priority value can preempt it.
```
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:556]:
```suggestion
This annotation specifies the minimum priority value that can preempt the
[PriorityClass](/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass).
It defaults to the PriorityClass's priority value +1, meaning that Pods with a higher priority value can preempt it.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 5]
```diff
@@ -547,6 +547,22 @@ This annotation requires the [NodePreferAvoidPods scheduling plugin](/docs/refer
...
+
+### preemption-toleration.scheduling.sigs.k8s.io/toleration-seconds
+
+Example: `preemption-toleration.scheduling.sigs.k8s.io/toleration-seconds: "3600"`
+
+Used on: PriorityClass
+
+This annotation specifies how long the [PriorityClass](/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass) can tolerate preemption by priorities lower than `preemption-toleration.scheduling.sigs.k8s.io/minimum-preemptable-priority`. It defaults to value 0, means the pod will be preempted immediately that is, no toleration.If set to a positive value, the duration will be honored. If set to a negative value, the pod can be tolerated forever that is, pods with a priority lower than minimum-preemptable-priority won't be able to preempt. This affects scheduled pods only (no effect on nominated pods).
```
Comment (tengqm) [inline content/en/docs/reference/labels-annotations-taints/_index.md:564]:
```suggestion
This annotation specifies how long the [PriorityClass](/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass)
can tolerate preemption by priorities lower than `preemption-toleration.scheduling.sigs.k8s.io/minimum-preemptable-priority`.
It defaults to "0", meaning that the Pod will be preempted immediately. That is, no toleration.
If set to a positive value, the duration will be honored. If set to a negative value, the Pod can be
tolerated forever. That is, Pods with a priority lower than `minimum-preemptable-priority` won't
be able to preempt. This affects scheduled Pods only (no effect on nominated Pods).
```

Comment (nitishchauhan0022) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: PTAL @tengqm

**Comment by kbhawkey** [comment]
@kubernetes/sig-scheduling-misc

**Comment by alculquicondor** [comment]
/hold
unclear if we should be documenting subprojects behavior in the main kubernetes website.

**Comment by sftim** [comment]
@alculquicondor, how can we decide when it's OK to unhold? It's good to explain that if we can.

**Comment by alculquicondor** [comment]
Let's discuss that in the issue.

**Comment by divya-mohan0209** [comment]
Per [this comment](https://github.com/kubernetes/website/issues/37589#issuecomment-1486854207), closing the PR.

/close

**Pull request closed without merging**